src/validation/rules/known_operation_types.rs

```diff
@@ -0,0 +1,140 @@
+use super::ValidationRule;
+use crate::ast::{    visit_document, OperationVisitor, OperationVisitorContext, SchemaDocumentExtension,
+    };
+use crate::static_graphql::query::*;
+use crate::validation::utils::{ValidationError, ValidationErrorContext};
+
+/// Known operation types
+///
+/// A GraphQL operation is only valid if the operation type is within the schema.
+///
+/// See https://github.com/graphql/graphql-spec/pull/947
+pub struct KnownOperationTypes;
+
+impl KnownOperationTypes {
+    pub fn new() -> Self {
+        KnownOperationTypes
+    }
+}
+
+fn build_error_message(root_type_name: &str) -> String {
+  format!("The {} operation is not supported by the schema.", root_type_name)
+}
+
+impl<'a> OperationVisitor<'a, ValidationErrorContext> for KnownOperationTypes {
+    fn enter_operation_definition(
+        &mut self,
+        visitor_context: &mut OperationVisitorContext,
+        user_context: &mut ValidationErrorContext,
+        operation_definition: &OperationDefinition,
+    ) {
+        match operation_definition {
+            OperationDefinition::Mutation(mutation) => {
+                if let None = visitor_context.schema.mutation_type() {
+                    user_context.report_error(ValidationError {
+                        locations: vec![mutation.position],
+                        message: build_error_message("mutation"),
+                    });
+                }
+            },
+            OperationDefinition::Subscription(subscription) => {
+                if let None = visitor_context.schema.subscription_type() {
+                    user_context.report_error(ValidationError {
+                        locations: vec![subscription.position],
+                        message: build_error_message("subscription"),
+                    });
+                }
+            },
+            OperationDefinition::SelectionSet(_) => {},
+            OperationDefinition::Query(_) => {},
```

Comment:
not sure whether there is a better way of handling this...

I tried `_ => {}`, which the linter did not like 👀

Reply:
`_ => {}` should work fine.... what linter did not like it?

Reply:
nevermind, I am tripping

Reply:
pushed the change